apps/src/lib/levelbuilder/course-editor/CourseEditor.js

```diff
@@ -134,28 +134,20 @@ export default class CourseEditor extends Component {
             style={styles.input}
           />
         </label>
-        <label>
-          Student Description
-          <textarea
-            name="description_student"
-            defaultValue={this.state.descriptionStudent}
-            rows={5}
-            style={styles.input}
-            onChange={this.handleStudentDescriptionChange}
-          />
-          <MarkdownPreview markdown={this.state.descriptionStudent} />
-        </label>
-        <label>
-          Teacher Description
-          <textarea
-            name="description_teacher"
-            defaultValue={this.state.descriptionTeacher}
-            rows={5}
-            style={styles.input}
-            onChange={this.handleTeacherDescriptionChange}
-          />
-          <MarkdownPreview markdown={this.state.descriptionTeacher} />
-        </label>
+        <TextareaWithMarkdownPreview
+          markdown={this.state.descriptionStudent}
```

Comment:
Should the `TextareaWithMarkdownPreview` component also own the markdown state and the onChange handler?

Reply:
Great idea! I have updated with that change